wordpress-mobile/WordPress-Android
# Jetpack focus: Disable push notifications for WP

If an instance of JP is detected then disable push notifications for WP to prevent double notifications

- [x] Trigger this from Jetpack app, on notifications screen in the welcome flow in a separate PR ( [See example](https://github.com/wordpress-mobile/WordPress-Android/pull/17400/commits/4b6c7caa12eea6d1ff8486fc43d36bd9fc5d4244) )
- [x] Add permissions to exported BroadcastReceiver.  Preferably, `signature`, otherwise `normal` in a separate PR

To test:

Test 1:
- Ensure either Jetpack app or Jetpack beta app is installed
- Ensure WordPress app is also on the device
- Ensure you already receive some push notifications

Test 2:
- Remove any WordPress app instances
- Build and Run WordPress app or install using QR code below
- Open Terminal in AS or Terminal app. Execute   `adb shell am broadcast -p org.wordpress.android.prealpha -a org.wordpress.android.broadcast.DISABLE_NOTIFICATIONS` command ([Welcome flow ](https://github.com/wordpress-mobile/WordPress-Android/pulls/ovitrif)is still in the works, till then use command line to test it on WP side)
- Verify `Action: org.wordpress.android.broadcast.DISABLE_NOTIFICATIONS` is displayed in a toast message on the app.  The same can be seen in the Logcat in AS
- Verify switch is set to Off on Notification Settings ( Me > Notifications tab > ⚙️ on the top right hand corner )
- Wait till the next day if required or [push a debug notification](https://mc.a8c.com/mobile/push-notifications/debug/)
- Verify push notifications are **not** displayed for WordPress app
- Ensure notifications are displayed for Jetpack app as before.


NOTE:  this doesn't cover any local notifications e.g. WeeklyRoundup


## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually

3. What automated tests I added (or what prevented me from doing so)
Existing tests

PR submission checklist:
